### ISSUE TITLE
Noisy Validation Accuracy

### ISSUE THREAD

**Issue by bkj** [fchollet/keras#2718]
Does anyone have any ideas as to why the validation accuracy curve would be this noisy?

![screen shot 2016-05-13 at 10 22 04 am](https://cloud.githubusercontent.com/assets/6086781/15250847/92ada084-18f4-11e6-9527-b0871e776065.png)

For completeness, I'll mention that the "validation set" is systematically different than the training set (this is a transfer learning situation).  I can provide more details as needed about the specific application and model, but was interested if there was some kind of generic explanation for the noisy accuracy curve.

**Comment by dansbecker** [comment]
Small validation set?

Also, accuracy is a funny metric in that it changes discontinuously as a function of the underlying predicted probabilities. It cross-entropy loss this noisy too?

**Comment by akilat90** [comment]
@bkj Would you mind telling how did you get the accuracy vector to get this curve plotted? I guess you've referred to an attribute of the model.fit() history object.
Ideally, I want to run a couple epochs and get the average of the validation accuracy for those epochs.

And for your question, as @dansbecker said, the problem might have a thing to do with the inadequacy  of validation set size. 
If you're using stochastic gradient descent, a small batch size might cause noise with regards to the convergence of the cost function, subsequently this will affect the accuracy as well. Here's a video by Andrew Ng explaining the same https://www.youtube.com/watch?v=y-xIqvhf8T4

**Comment by bkj** [comment]
I believe (IIRC) that it was due to not shuffling the data -- I was using
fit_generator, which doesn't always lend itself to shuffling, so the
peaks/valleys of error rate correspond to the same sets of records in the
data. Possibly the way my data was sorted was also nonrandom, but I didn't
look into that very deeply.
